### PR TITLE
CREATE DATABASE ... CONNECTION LIMIT

### LINKED ISSUE #8501: release-20.2: sql: allow CONNECTION LIMIT syntax for CREATE DATABASE
PR: https://github.com/cockroachdb/cockroach/pull/54421

From release notes:
> Added the ability to parse the `CONNECTION LIMIT [=] iconst32` syntax for [`CREATE DATABASE`](../v20.2/create-database.html). This is currently a no-op for -1, and errors for any other value.  By default, the value is -1 (unlimited connections). [#54421][#54421] {% comment %}doc{% endcomment %}

### PR DESCRIPTION
**Pull request by ericharmeling**
Fixes #8501.

### CODE REVIEW

File: v20.2/create-database.md [hunk 1]
```diff
@@ -25,6 +25,7 @@ Parameter | Description
 `IF NOT EXISTS` | Create a new database only if a database of the same name does not already exist; if one does exist, do not return an error.
 `name` | The name of the database to create, which [must be unique](#create-fails-name-already-in-use) and follow these [identifier rules](keywords-and-identifiers.html#identifiers).
 `encoding` | The `CREATE DATABASE` statement accepts an optional `ENCODING` clause for compatibility with PostgreSQL, but `UTF-8` is the only supported encoding. The aliases `UTF8` and `UNICODE` are also accepted. Values should be enclosed in single quotes and are case-insensitive.<br><br>Example: `CREATE DATABASE bank ENCODING = 'UTF-8'`.
+`CONNECTION LIMIT` | <span class="version-tag">New in v20.2:</span> Supported for compatibility with PostgreSQL. By default, CockroachDB does not limit connections, and `CONNECTION LIMIT = -1`. This clause is a no-op for the default value (`-1`), and all other values return an error.
```
Comment (otan) [inline v20.2/create-database.md:28]: nit: ```` , and `CONNECTION_LIMIT=1` ``` is worded in a way hard to read. Maybe `A value of -1 indicates no limit. By default, CockroachDB sets `CONNECTION LIMIT = -1`

Comment (ericharmeling) [reply v20.2/create-database.md]: TFTR, @otan. You're right - it was a little lacking in clarity. Great suggestions. Updated.

**Review by otan — APPROVED** [review]

**Review by lnhsingh — COMMENTED** [review]
LGTM, one nit/question

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/8670#-:-MJi2wIr8uiHDKLU3Bh-:b2u0osj)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @ericharmeling, @lnhsingh, and @otan)

___
*[v20.2/create-database.md, line 28 at r2](https://reviewable.io/reviews/cockroachdb/docs/8670#-MJi2sMo0mC-3JtII06_:-MJi2sMo0mC-3JtII06a:b93jogf) ([raw file](https://github.com/cockroachdb/docs/blob/c2a92c8e3e6dc3766104b5df9bae6f4a22c86c12/v20.2/create-database.md#L28)):*
> ```Markdown
> `name` | The name of the database to create, which [must be unique](#create-fails-name-already-in-use) and follow these [identifier rules](keywords-and-identifiers.html#identifiers).
> `encoding` | The `CREATE DATABASE` statement accepts an optional `ENCODING` clause for compatibility with PostgreSQL, but `UTF-8` is the only supported encoding. The aliases `UTF8` and `UNICODE` are also accepted. Values should be enclosed in single quotes and are case-insensitive.<br><br>Example: `CREATE DATABASE bank ENCODING = 'UTF-8'`.
> `CONNECTION LIMIT` | <span class="version-tag">New in v20.2:</span> Supported for compatibility with PostgreSQL. A value of -1 indicates no connection limit. Values other than -1 are currently not supported. By default, `CONNECTION LIMIT = -1`. 
> ```

nit: Should -1 be `-1`?

**Review by ericharmeling — COMMENTED** [review]
TFTR @lnhsingh !

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/8670#-:-MK-xral6jTSxn0fIJcl:bhjhpnt)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @lnhsingh and @otan)

___
*[v20.2/create-database.md, line 28 at r2](https://reviewable.io/reviews/cockroachdb/docs/8670#-MJi2sMo0mC-3JtII06_:-MK-xqVI9pM0Agzm1Xuh:bt2vr1b) ([raw file](https://github.com/cockroachdb/docs/blob/c2a92c8e3e6dc3766104b5df9bae6f4a22c86c12/v20.2/create-database.md#L28)):*
<details><summary><i>Previously, lnhsingh (Lauren Hirata Singh) wrote…</i></summary><blockquote>

nit: Should -1 be `-1`?
</blockquote></details>

Fixed.

**Pull request merged**